Source/App/EbAppContext.c

```diff
@@ -220,6 +220,11 @@ EB_ERRORTYPE CopyConfigurationParameters(
         printf("\nWarning: input profile is not correct, force converting it from %d to MainREXT for YUV422 or YUV444 cases \n", config->profile);
         callbackData->ebEncParameters.profile = 4;
     }
+    else if(config->encoderBitDepth > 8 && config->profile < 2)
+    {
+        printf("\nWarning: input profile is not correct, force converting it from %d to Main10 for 10 bits cases\n", config->profile);
+        callbackData->ebEncParameters.profile = 2;
+    }
```

Comment:
Hi Jing, I didn't see similar logic to decide main/main10 based on bitdepth and colorformat in ffmpeg plugin? I see the logic to decide mainext. What did I miss?

Reply:
Hi @jsunintel , any update?

Reply:
It's updated last week following your comments. Please review and merge it if no more issue.